src/main/webapp/resources/js/pages/clients/listing/ClientListingPage.jsx

```diff
@@ -0,0 +1,33 @@
+import React from "react";
+import { render } from "react-dom";
+import { PageWrapper } from "../../../components/page/PageWrapper";
+import { PagedTableProvider } from "../../../components/ant.design/PagedTable";
+import { ClientsTable } from "./ClientsTables";
+import { setBaseUrl } from "../../../utilities/url-utilities";
+import { Button } from "antd";
+import { PlusCircleTwoTone } from "@ant-design/icons";
+
+/**
+ * Page for displaying the list of all clients.
+ * @return {*}
+ * @constructor
+ */
+function ClientListingPage() {
+  return (
+    <PageWrapper
+      title={i18n("clients.title")}
+      headerExtras={
+        <Button type={"primary"} href={setBaseUrl(`clients/create`)}>
+          <PlusCircleTwoTone />
+          {i18n("clients.add")}
+        </Button>
+      }
+    >
+      <PagedTableProvider url={setBaseUrl("clients/ajax/list")}>
```

Comment:
Same in this class, will these URLs work in a context path?

Reply:
Same as above